community/kernel/src/main/java/org/neo4j/kernel/impl/api/LockingStatementOperations.java

```diff
@@ -266,6 +257,21 @@ public void visit( long relId, int type, long startNode, long endNode )
         entityWriteDelegate.relationshipDelete( state, relationshipId );
     }
 
+    private void lockRelationshipNodes( KernelStatement state, long startNodeId, long endNodeId )
+    {
+        // Order the locks to lower the risk of deadlocks with other threads adding rels concurrently
+        if ( startNodeId < endNodeId )
```

Comment:
Maybe we should have another branch for `startNodeId == endNodeId` where we only take the lock once?